src/modules/map/components/cesium/CesiumGPXLayer.vue

```diff
@@ -0,0 +1,159 @@
+<script setup>
+import {
+    BillboardGraphics,
+    Cartesian3,
+    Color,
+    ColorMaterialProperty,
+    defined as cesiumDefined,
+    GpxDataSource,
+    HeightReference,
+} from 'cesium'
+import { onMounted, onUnmounted, ref, watch } from 'vue'
+import { inject } from 'vue'
+
+import GPXLayer from '@/api/layers/GPXLayer.class'
+import log from '@/utils/logging'
+
+const props = defineProps({
+    gpxLayerConfig: {
+        type: GPXLayer,
+        required: true,
+    },
+})
+
+const isPresentOnMap = ref(false)
+const gpxDataSource = ref(new GpxDataSource())
+
+const opacity = ref(props.gpxLayerConfig.opacity)
+const gpxData = ref(props.gpxLayerConfig.gpxData)
+
+const getViewer = inject('getViewer')
+watch(gpxData, () => {
+    addLayer()
+})
```

Comment:
If the layer was already added, I think it would make to also remove it before adding it again.
I don't think what `addLayer` does removes the "old" data